### PR TITLE
read only remote map creates db only if files not exist

### PR DESCRIPTION
**Pull request by noamhammer**
_read only remote map creates db only if files not exist_

---

_Additional context for reviewer_

---
_Release Notes_: 
_None_

---

### CODE REVIEW

File: packages/core/src/local-workspace/remote_map.ts [hunk 1]
```diff
@@ -577,6 +561,21 @@ remoteMap.RemoteMapCreator => {
       delKeys.add(key)
       locationCache.del(key)
     }
+    const createDBIfNotExist = async (loc: string): Promise<void> => {
+      const newDb: rocksdb = getRemoteDbImpl()(loc)
+      try {
+        await promisify(newDb.open.bind(newDb, { readOnly: true }))()
```
Comment (roironn) [inline packages/core/src/local-workspace/remote_map.ts:567]: Not sure that there is a need to create a read connection here

Comment (roironn) [reply packages/core/src/local-workspace/remote_map.ts:567]: Also - can we choose another name for this function and the other function named `createDBIfNotExist`?

This is confusing :D

Comment (roironn) [reply packages/core/src/local-workspace/remote_map.ts:567]: Last one - do we even need this function considering that we are creating a write connection which will create the db anyway?

File: packages/core/src/local-workspace/remote_map.ts [hunk 2]
```diff
@@ -725,6 +724,19 @@ remoteMap.ReadOnlyRemoteMapCreator => {
       }
       return createIterator(keyPrefix, normalizedOpts, db)
     }
+    const createDBIfNotExist = async (loc: string): Promise<void> => {
+      const newDb: rocksdb = getRemoteDbImpl()(loc)
+      if (await fileUtils.isEmptyDir.notFoundAsUndefined(loc) === false) {
```
Comment (roironn) [inline packages/core/src/local-workspace/remote_map.ts:729]: I think it would be better to check for rocksdb files in the root dir. For example, assume we create a remote map at a location, and from some reason destroy only the main db (which means that the tmpdb folders will remain) - the directory will not be empty but the db is missing.

Also - this type of logic (and the edge case mentioned above needs to be tests :) )

File: packages/core/src/local-workspace/remote_map.ts [hunk 3]
```diff
@@ -733,7 +748,7 @@ remoteMap.ReadOnlyRemoteMapCreator => {
       }
       const connectionPromise = (async () => {
         currentConnectionsCount += 1
-        await createDBIfNotExist(location)
+        await checkDBExists(location)
```
Comment (roironn) [inline packages/core/src/local-workspace/remote_map.ts:751]: Why do we need this? Can't we just try and catch when we create the read only connection? I don't think we need the extra open -> close flow here...

File: packages/core/src/local-workspace/remote_map.ts [hunk 4]
```diff
@@ -577,6 +561,28 @@ remoteMap.RemoteMapCreator => {
       delKeys.add(key)
       locationCache.del(key)
     }
+    const createDBIfNotExist = async (loc: string): Promise<void> => {
+      const newDb: rocksdb = getRemoteDbImpl()(loc)
+      const readOnly = !persistent
+      try {
+        await promisify(newDb.open.bind(newDb, { readOnly }))()
```
Comment (roironn) [inline packages/core/src/local-workspace/remote_map.ts:568]: Why do we need to open the read connection? Won't creating a write connection if the DB does not exists create the DB?

Comment (noamhammer) [reply packages/core/src/local-workspace/remote_map.ts:568]: because if the DB does exist and the remote map is in non-persistent mode, I don't want to create a write connection to the DB

**Review by roironn — APPROVED** [review]
